src/cmd/init.rs

```diff
@@ -223,18 +264,21 @@ pub fn run_init(wallet_dir: &std::path::Path, cmd: InitOpt) -> anyhow::Result<Cm
                 MemoryDatabase::default(),
             )?;
 
+            let signers = vec![GunSigner::PsbtSdCard { path: psbt_dir }];
+
             Config {
-                wallet_key: WalletKey::Descriptor { external, internal },
-                psbt_output_dir: psbt_dir,
-                ..Config::default_config(common_args.network)
+                descriptor_external: external.clone(),
+                descriptor_internal: internal,
+                signers,
+                ..Config::default_config(common_args.network, external)
             }
         }
         InitOpt::XPub {
             common_args,
-            psbt_output_dir,
+            psbt_signer_dir,
             ref xpub,
         } => {
-            let psbt_dir = create_psbt_dir(wallet_dir, psbt_output_dir)?;
+            let psbt_dir = create_psbt_dir(wallet_dir, psbt_signer_dir)?;
```

Comment:
Currently if `psbt_signer_dir` is not specified, a `/psbts` directory is created and used. Users who `init` with `xpub` or `descriptor` could still sign these transactions and broadcast, so isn't just watch-only.

Changed so that the directory is only created if none is specified (only creates `/psbt`/)